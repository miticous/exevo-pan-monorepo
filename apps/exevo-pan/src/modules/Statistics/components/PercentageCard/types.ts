export interface PercentageCardProps
  extends React.HTMLAttributes<HTMLDivElement> {
  title: string
  percentage: number
}

export interface PercentageStyleProps {
  positive: boolean
}
