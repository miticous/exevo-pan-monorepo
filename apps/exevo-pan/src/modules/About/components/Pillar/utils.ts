export const generateNavId = (id?: string): string => `${id}-nav`
