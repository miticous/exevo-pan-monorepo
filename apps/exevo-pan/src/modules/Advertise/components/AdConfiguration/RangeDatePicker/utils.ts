const formatDate = (timestamp: number): string => {
  const date = new Date(timestamp)
  const day = date.getDate().toString()
  const month = (date.getMonth() + 1).toString()
  const year = date.getFullYear().toString()

  return `${month}/${day}/${year}`
}

export const getDaysUntilAuctionEnd = (auctionEnd?: number): string[] => {
  if (!auctionEnd) return []

  const formatedEndDate = formatDate(auctionEnd * 1000)
  const formatedCurrentDate = formatDate(+new Date())

  const endDate = new Date(formatedEndDate)
  const currentDate = new Date(formatedCurrentDate)
  const daysInRange = [formatedCurrentDate]
  while (currentDate < endDate) {
    currentDate.setDate(currentDate.getDate() + 1)
    daysInRange.push(formatDate(+currentDate))
  }

  return daysInRange
}

export const partitionByMonths = (dateArray: string[]): string[][] => {
  const monthsPartition = new Set<string>([])

  dateArray.forEach((dateString) => {
    const [month] = dateString.split('/')
    monthsPartition.add(month)
  })

  return [...monthsPartition].map((month) =>
    dateArray.filter((dateString) => {
      const [dateStringMonth] = dateString.split('/')

      return month === dateStringMonth
    }),
  )
}

export const fillWithDays = (
  initialDateString: string,
  amount: number,
  step: number,
): string[] => {
  const currentDate = new Date(initialDateString)

  return Array.from(
    { length: amount },
    () => +currentDate.setDate(currentDate.getDate() + step),
  )
    .sort((a, b) => a - b)
    .map(formatDate)
}

export const getWeekdayNumber = (date: string): number =>
  new Date(date).getDay()

export const getDay = (date: string): number => new Date(date).getDate()

export const getMonth = (date: string): number => new Date(date).getMonth()
