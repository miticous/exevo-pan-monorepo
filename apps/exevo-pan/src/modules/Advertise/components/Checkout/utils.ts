import { endpoints } from 'Constants'

const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/
export const validateEmail = (email: string): boolean => emailRegex.test(email)

const buildUrl = (nickname: string): string =>
  `${endpoints.TIBIADATA}/${nickname}.json`

export const validateCharacter = async (nickname: string): Promise<boolean> => {
  try {
    const response = await fetch(buildUrl(nickname))
    const data = await response.json()
    return !data.characters.error
  } catch (error) {
    return true
  }
}

export const randomCharacter = (): string => {
  const characters = [
    'Eternal Oblivion',
    'Cachero',
    'Bubble',
    'Setzer Gambler',
    'Arieswar',
    'Kharsek',
    "Lord'Paulistinha",
    'â€ŽMateusz Dragon Wielki',
  ]

  return characters[Math.floor(Math.random() * characters.length)]
}
