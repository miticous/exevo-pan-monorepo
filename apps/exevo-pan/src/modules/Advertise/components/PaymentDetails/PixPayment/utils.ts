import { QrCodePix } from 'qrcode-pix'
import { advertising } from 'Constants'

interface PixParams {
  txId: number
  daysAmount: number
}

export const generateQrCode = async ({
  txId,
  daysAmount,
}: PixParams): Promise<PixObject> => {
  const id = txId.toString()
  const tx = QrCodePix({
    version: '01',
    key: advertising.PIX_KEY,
    name: id,
    city: id,
    message: id,
    transactionId: id,
    value: daysAmount * advertising.BRL_ADVERTISE,
  })

  return {
    payload: tx.payload(),
    qrCode: await tx.base64(),
  }
}
