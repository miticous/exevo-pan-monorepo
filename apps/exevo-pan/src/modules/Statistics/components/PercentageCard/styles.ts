import styled from 'styled-components'
import { MaterialCard, Smooth } from 'styles'
import { PercentageStyleProps } from './types'

export const Wrapper = styled.section`
  ${MaterialCard}
  padding: 20px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  ${Smooth}
`

export const Title = styled.h4`
  font-size: 16px;
  font-weight: 300;
  color: var(--onSurface);
`

export const Percentage = styled.span<PercentageStyleProps>`
  font-size: 46px;
  font-weight: 700;
  color: ${({ positive }) => (positive ? 'var(--green)' : 'var(--red)')};
`
