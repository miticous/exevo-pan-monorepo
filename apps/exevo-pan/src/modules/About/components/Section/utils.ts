import { copyToClipboard } from 'utils'

export const CopyToClipboard = (id: string): void =>
  copyToClipboard(
    `https://${window.location.hostname}${window.location.pathname}#${id}`,
  )
