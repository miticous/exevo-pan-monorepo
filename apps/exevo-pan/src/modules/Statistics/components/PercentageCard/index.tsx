import { memo } from 'react'
import * as S from './styles'
import { PercentageCardProps } from './types'

const PercentageCard = ({
  title,
  percentage,
  ...props
}: PercentageCardProps): JSX.Element => (
  <S.Wrapper {...props}>
    <S.Title>{title}</S.Title>
    <S.Percentage positive={percentage >= 50}>{percentage}%</S.Percentage>
  </S.Wrapper>
)

export default memo(PercentageCard)
