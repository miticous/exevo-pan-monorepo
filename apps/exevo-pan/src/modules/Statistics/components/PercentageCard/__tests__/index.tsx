import { screen } from '@testing-library/react'
import { renderWithProviders } from 'utils/test'
import PercentageCard from '..'

describe('<PercentageCard />', () => {
  test('should render every text content', () => {
    renderWithProviders(
      <PercentageCard title="Auction success rate" percentage={55.13} />,
    )

    expect(screen.getByText(/55\.13/)).toBeInTheDocument()
    expect(screen.getByRole('heading')).toHaveTextContent(
      'Auction success rate',
    )
  })

  test('should display values correctly', () => {
    const { rerender } = renderWithProviders(
      <PercentageCard title="Auction success rate" percentage={0} />,
    )

    expect(screen.getByText(/0/)).toBeInTheDocument()

    rerender(<PercentageCard title="Auction success rate" percentage={1} />)
    expect(screen.getByText(/1/)).toBeInTheDocument()

    rerender(<PercentageCard title="Auction success rate" percentage={10} />)
    expect(screen.getByText(/10/)).toBeInTheDocument()

    rerender(<PercentageCard title="Auction success rate" percentage={0.1} />)
    expect(screen.getByText(/0\.1/)).toBeInTheDocument()

    rerender(<PercentageCard title="Auction success rate" percentage={100} />)
    expect(screen.getByText(/100/)).toBeInTheDocument()

    rerender(<PercentageCard title="Auction success rate" percentage={1000} />)
    expect(screen.getByText(/1000/)).toBeInTheDocument()
  })

  test('should style values correctly', () => {
    const { rerender } = renderWithProviders(
      <PercentageCard title="Auction success rate" percentage={0} />,
    )

    const percentageElement = screen.getByText(/0/)
    /* expect(percentageElement).toHaveStyle('color: #C51313;') */
    expect(percentageElement).toMatchSnapshot()

    rerender(<PercentageCard title="Auction success rate" percentage={50} />)
    /* expect(percentageElement).toHaveStyle('color: #377712;') */
    expect(percentageElement).toMatchSnapshot()

    rerender(<PercentageCard title="Auction success rate" percentage={100} />)
    /* expect(percentageElement).toHaveStyle('color: #377712;') */
    expect(percentageElement).toMatchSnapshot()

    rerender(<PercentageCard title="Auction success rate" percentage={1000} />)
    /* expect(percentageElement).toHaveStyle('color: #377712;') */
    expect(percentageElement).toMatchSnapshot()

    rerender(<PercentageCard title="Auction success rate" percentage={0.1} />)
    /* expect(percentageElement).toHaveStyle('color: #C51313;') */
    expect(percentageElement).toMatchSnapshot()

    rerender(
      <PercentageCard title="Auction success rate" percentage={49.9999} />,
    )
    /* expect(percentageElement).toHaveStyle('color: #C51313;') */
    expect(percentageElement).toMatchSnapshot()
  })
})
