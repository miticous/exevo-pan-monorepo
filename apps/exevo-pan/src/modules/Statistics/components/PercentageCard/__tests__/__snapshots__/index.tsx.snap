// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PercentageCard /> should style values correctly 1`] = `
<span
  class="styles__Percentage-sc-5s1rht-2 kASbqQ"
>
  0
  %
</span>
`;

exports[`<PercentageCard /> should style values correctly 2`] = `
<span
  class="styles__Percentage-sc-5s1rht-2 hxmePE"
>
  50
  %
</span>
`;

exports[`<PercentageCard /> should style values correctly 3`] = `
<span
  class="styles__Percentage-sc-5s1rht-2 hxmePE"
>
  100
  %
</span>
`;

exports[`<PercentageCard /> should style values correctly 4`] = `
<span
  class="styles__Percentage-sc-5s1rht-2 hxmePE"
>
  1000
  %
</span>
`;

exports[`<PercentageCard /> should style values correctly 5`] = `
<span
  class="styles__Percentage-sc-5s1rht-2 kASbqQ"
>
  0.1
  %
</span>
`;

exports[`<PercentageCard /> should style values correctly 6`] = `
<span
  class="styles__Percentage-sc-5s1rht-2 kASbqQ"
>
  49.9999
  %
</span>
`;
